src/GeneratorV2/Process/Processor.cs

```diff
@@ -475,7 +477,8 @@ public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List
                     var returnType = new CSString(Nullable: true);
                     newOverloads = new List<Overload>()
                     {
-                        new Overload(overload, layer, overload.InputParameters, overload.NativeFunction, returnType, newReturnName, overload.GenericTypes)
+                        overload with{NestedOverload =  overload, MarshalLayerToNested = layer, ReturnType = returnType, ReturnVariableName = newReturnName},
```

Comment:
Minor format detail but the Microsoft style puts a space between `with` and `{`.